lib/internal/http2/compat.js

```diff
@@ -234,6 +235,10 @@ class Http2ServerRequest extends Readable {
     return this[kHeaders][HTTP2_HEADER_METHOD];
   }
 
+  set method(method) {
+    this[kHeaders][HTTP2_HEADER_METHOD] = method;
```

Comment:
should this include a basic type check to ensure that `method` is a non-empty string?

Reply:
Good point. http1 obviously didn't have it since it's not a getter/setter but I don't see why we couldn't do that for http2.